File: frontend/src/lib/mapStyles.ts
// Shared basemap style URLs for every MapLibre view.
//
// These were three separate inline raster style objects until 2026-09-01,
// when CARTO began stamping an "API KEY REQUIRED" watermark across its
// unauthenticated raster tiles (basemaps.cartocdn.com/dark_all/...) and
// every dark basemap in the app broke at once.
//
// OpenFreeMap replaces them: OpenMapTiles-schema vector tiles served with
// no API key, no signup and no request quota, and it publishes a
// dark/light pair that drops straight onto the existing per-view toggle.
// Vector also renders sharper than the old raster tiles on high-DPI
// displays. Attribution travels in the TileJSON, so MapLibre renders the
// OpenFreeMap / OpenMapTiles / OpenStreetMap credits on its own — do not
// add an AttributionControl for it.
//
// The trade-off is that OpenFreeMap is a free public instance with no
// SLA. If it ever goes away, the migration path is a self-hosted
// Protomaps .pmtiles extract served from the stack's own Caddy container.
export const DARK_STYLE = "https://tiles.openfreemap.org/styles/dark";
export const LIGHT_STYLE = "https://tiles.openfreemap.org/styles/positron";

File: android/app/src/main/kotlin/com/pitstop/ui/history/MapStyles.kt
package com.pitstop.ui.history

/**
 * Basemap style URLs shared by the route and heatmap maps.
 *
 * Both maps ran on CARTO's Dark Matter GL style until 2026-09-01, when
 * CARTO began stamping an "API KEY REQUIRED" watermark across its
 * unauthenticated raster tiles. The vector style the phone used was not
 * watermarked at the time, but CARTO is retiring the free tier, so the
 * phone moved to OpenFreeMap alongside the web frontend rather than wait
 * for the same break — see frontend/src/lib/mapStyles.ts, which this file
 * mirrors for web/phone parity.
 *
 * OpenFreeMap serves OpenMapTiles-schema vector tiles with no API key, no
 * signup and no request quota. Attribution travels in the TileJSON, so
 * MapLibre renders the credits on its own once isAttributionEnabled is set.
 */
internal object MapStyles {
    /** Trip-detail and heatmap maps are both fixed dark on the phone. */
    const val DARK = "https://tiles.openfreemap.org/styles/dark"

    /** Light counterpart, for parity with the web toggle. Unused for now. */
    const val LIGHT = "https://tiles.openfreemap.org/styles/positron"
}
